### ISSUE TITLE
Can't transform Plane as it lacks `operator *`

### ISSUE THREAD

**Issue by ywmaa** [godotengine/godot#63489]
### Godot version

4.0.alpha12

### System information

Pop!OS 22.04

### Issue description

previous godot engine versions 3.x we had a function "xform" :
transform.xform(Plane)
, it was removed in favour of the operator *
but Planes weren't added to the operator *
most types that worked with "xform" before are now using the operator * except Plane it was not added with them.
so AABB, PackedVector3Array, Vector3 are there except the type "plane"

TLDR : I can't transform a Plane

### Steps to reproduce

Any project in Godot 4 alpha 12

### Minimal reproduction project

N/A

**Comment by sairam4123** [comment]
Can confirm!

**Comment by Chaosus** [comment]
`Vector4` lacks this operator too. Why not simply use `Quaternion` where that operation is supported?

**Comment by ywmaa** [comment]
how I should transform a plane using a `Quaternion` can you provide and example, pseudo code ?

**Comment by kleonc** [comment]
> Why not simply use `Quaternion` where that operation is supported?

For the same reason you'd use `Transform` instead of `Quaternion` anywhere else. You can't e.g. translate or non-uniformly scale a plane using a quaternion.
